src/main/scala/better/files/package.scala

```diff
@@ -59,9 +59,16 @@ package object files {
 
     def /(f: File => File): File = f(this)
 
-    def createChild(child: String): File = (this / child).createIfNotExists()
+    def createChild(child: String, asDirectory: Boolean = false): File = (this / child).createIfNotExists(asDirectory)
 
-    def createIfNotExists(): File = if (exists) this else {parent.createDirectories(); Files.createFile(path)}
+    def createIfNotExists(asDirectory: Boolean = false): File = if (exists) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Consider using case matching instead of else if blocks](https://www.codacy.com/app/pathikrit/better-files/file/1018061947/issues/source?bid=2860326&fileBranchId=2876420#l64)